viewing_party/party.py

```diff
@@ -0,0 +1,193 @@
+def create_movie(title, genre, rating):
+    """
+    This function returns a dictionary that gives information about a
+    movie's title, genre and rating when all 3 parameters are truthy. 
+    Otherwise, the function returns None.
+    """
+    new_movie = {}
+    if title and genre and rating:
+        new_movie["title"] = "Title A"
+        new_movie["genre"] = "Horror"
+        new_movie["rating"] = 3.5
+        return new_movie
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    """  
+    The function returns a dictionary that gives information about the 
+    movies that the user has watched. This function adds another 
+    dictionary, movie, to the list of dictionaries in user_data. 
+    """
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    """
+    The function returns a dictionary that gives information about a 
+    user's watchlist. This function adds another dictionary, movie, 
+    to the list of dictionaries in user_data.
+    """
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title): 
+    """
+    The function returns a dictionary that gives information about 
+    movies that the user has watched and movies that are on the 
+    user's watchlist. If the title is in the user's watchlist, 
+    the function will remove the movie from the watchlist and add 
+    it to the list of movies that the user has already watched. 
+    Otherwise, the function will simply return user_data. 
+    """
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data
+    return user_data
+
+# Wave 2
+def get_watched_avg_rating(user_data):
+    """
+    The function calculates and returns the average rating from 
+    a list of movies that the user has watched. The function 
+    returns 0 if the user's watched list is empty.
+    """
+    watched_movies = user_data["watched"]
+    sum = 0
+    if len(watched_movies) > 0:
+        for movie in watched_movies:
+            sum += movie["rating"]
+        avg_rating = sum/len(watched_movies)
+        return avg_rating
+    return 0
+
+def get_most_watched_genre(user_data):
+    """
+    The function returns the genre that is most frequently watched 
+    by the user from a list of movies that the user has watched. 
+    The function returns None if the list of movies that the user 
+    has watched is empty. 
+    """
+    genre_frequency = {}
+    if len(user_data["watched"]) == 0:
+        return None
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre not in genre_frequency:
+            genre_frequency[genre] = 1
+        else:
+            genre_frequency[genre] += 1
+
+    for genre, frequency in genre_frequency.items():
+        if frequency  == max(genre_frequency.values()):
```

Comment:
I like that you split this up a bit more. it's definitely easier to read when it's not all one line